internal/services/v1/clustering_test.go

```diff
@@ -0,0 +1,288 @@
+package v1
+
+import (
+	"context"
+	"math"
+	"sort"
+	"testing"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/stretchr/testify/require"
+
+	"github.com/authzed/spicedb/internal/graph/computed"
+	"github.com/authzed/spicedb/pkg/datastore"
+	"github.com/authzed/spicedb/pkg/testutil"
+	"github.com/authzed/spicedb/pkg/tuple"
+)
+
+type expectedClusteredRequest struct {
+	resourceType string
+	resourceRel  string
+	subject      string
+	resourceIDs  [][]string
+}
+
+func TestClusterItems(t *testing.T) {
+	testCases := []struct {
+		name      string
+		requests  []string
+		response  []expectedClusteredRequest
+		chunkSize uint16
+		err       string
+	}{
+		{
+			name: "different subjects cannot be clustered",
```

Comment:
please add a test like I described above and ensure it doesn't accidentally cluster